HealthGPT/HealthGPT/HealthGPTView.swift

```diff
@@ -6,58 +6,59 @@
     @AppStorage(StorageKeys.enableTextToSpeech) private var textToSpeech = StorageKeys.Defaults.enableTextToSpeech
-    @EnvironmentObject private var openAPIComponent: OpenAIComponent
-    @EnvironmentObject private var healthDataInterpreter: HealthDataInterpreter
+    @AppStorage(StorageKeys.openAIModel) private var openAIModel = LLMOpenAIModelType.gpt4
+    
+    @Environment(HealthDataInterpreter.self) private var healthDataInterpreter
     @State private var showSettings = false
-    @StateObject private var speechSynthesizer = SpeechSynthesizer()
-
     
     var body: some View {
-        NavigationView {
-            VStack {
-                ChatView($healthDataInterpreter.runningPrompt, disableInput: $healthDataInterpreter.querying)
+        NavigationStack {
+            if let llm = healthDataInterpreter.llm {
+                let contextBinding = Binding { llm.context } set: { llm.context = $0 }
+                ChatView(contextBinding)
+                    .speak(llm.context, muted: !textToSpeech)
+                    .speechToolbarButton(muted: !$textToSpeech)
+                    .viewStateAlert(state: llm.state)
                     .navigationTitle("WELCOME_TITLE")
-                    .gesture(
-                        TapGesture().onEnded {
-                            UIApplication.shared.hideKeyboard()
-                        }
-                    )
                     .toolbar {
                         ToolbarItem(placement: .primaryAction) {
                             settingsButton
                         }
-                        ToolbarItem(placement: .primaryAction) {
-                            textToSpeechButton
+                    }
+                    .onChange(of: llm.context, initial: true) { _, _ in
+                        Task {
+                            if llm.state != .generating {
+                                try? await healthDataInterpreter.queryLLM()
+                            }
                         }
                     }
-            }
-            .onAppear {
-                generatePrompt()
-            }
-            .onChange(of: completedOnboardingFlow) { _ in
-                generatePrompt()
-            }
-            .onChange(of: healthDataInterpreter.querying) { _ in
-                if textToSpeech,
-                    healthDataInterpreter.runningPrompt.last?.role == .assistant,
-                    let lastMessageContent = healthDataInterpreter.runningPrompt.last?.content {
-                    speechSynthesizer.speak(lastMessageContent)
+            } else {
+                VStack {
+                    Text("LOADING_CHAT_VIEW")
+                    ProgressView()
                 }
             }
-            .sheet(isPresented: $showSettings) {
-                SettingsView(chat: $healthDataInterpreter.runningPrompt)
+        }
+        .sheet(isPresented: $showSettings) {
+            SettingsView()
+        }
+        .onAppear {
```

Comment:
Use the more modern `.task` modifier, then you can also perform `async` operations in the closure